### PR TITLE
Fix fork() without MMU

### PR DESCRIPTION
**Pull request by Deryugin**
Finally, fork() without MMU works :)

### CODE REVIEW

File: templates/x86/qemu/mods.config [hunk 1]
```diff
@@ -73,6 +74,8 @@ configuration conf {
 	@Runlevel(1) include embox.test.stdlib.qsort_test
 	@Runlevel(1) include embox.test.posix.environ_test
 	@Runlevel(1) include embox.test.posix.timerfd_test
+	//include embox.test.posix.forktest_fork_static
+	//@Runlevel(1) include embox.test.posix.fork_test
```
Comment (anton-bondarev) [inline templates/x86/qemu/mods.config]: is it work If it's uncommented?
may be add also **@Runlevel(1) include embox.test.posix.vfork_test**

File: src/kernel/thread/thread_switch.c [hunk 2]
```diff
@@ -33,13 +33,15 @@ static void thread_prepare_switch(struct thread *prev, struct thread *next) {
...
 
+extern void thread_set_current(struct thread *t);
 void thread_context_switch(struct thread *prev, struct thread *next) {
 	thread_prepare_switch(prev, next);
 
 	/* Preserve initial semantics of prev/next. */
 	cpudata_var(saved_prev) = prev;
 	ADDR_SPACE_PREPARE_SWITCH();
 
+	thread_set_current(next);
```
Comment (anton-bondarev) [inline src/kernel/thread/thread_switch.c]: May be move under **context_switch**

Comment (Deryugin) [reply src/kernel/thread/thread_switch.c]: `context_switch()` is platform-dependent function, so in this case we will need to duplicate this function call. Also changing current thread in `thread_context_switch()` doesn't seem wrong.

File: src/compat/posix/proc/fork_nommu/fork_addr_space_heap.c [hunk 3]
```diff
@@ -38,11 +38,12 @@ void fork_heap_store(struct heap_space *hpspc) {
 		hpspc->heap = phymem_alloc(size / PAGE_SIZE());
 		assert(hpspc->heap);
 	}
-	mspace_deep_store(task_mspace(), &hpspc->store_space, hpspc->heap);
+	mspace_deep_store(task_mspace(tk), &hpspc->store_space, hpspc->heap);
+	//mspace_deep_store(task_self_mspace(), &hpspc->store_space, hpspc->heap);
```
Comment (anton-bondarev) [inline src/compat/posix/proc/fork_nommu/fork_addr_space_heap.c]: Remove commented code, please

File: src/drivers/tty/termios_ops.c [hunk 4]
```diff
@@ -162,10 +138,10 @@ static int termios_handle_erase(const struct termios *t, char ch,
 }
 
 static int termios_input_status(const struct termios *t,
-	struct termios_i_buff *b, char ch) {
-	
+	struct termios_i_buff *b, char ch, int is_eol) {
+
 	int raw_mode = !TIO_L(t, ICANON);
-	int is_eol = (ch == '\n' || ch == t->c_cc[VEOL]);
+	//int is_eol = (ch == '\n' || ch == t->c_cc[VEOL]);
```
Comment (anton-bondarev) [inline src/drivers/tty/termios_ops.c]: Remove commented code, please

**Comment by Deryugin** [comment]
Outdated diff comments were fixed.

**Pull request merged**